### PR TITLE
Fix for more than 2 tunneldigger brokers:

### PR DESCRIPTION
**Pull request by Fungur69**
select() moved from context_process() to a new function do_select(),
so that packets get waited for for all brokers at once. After that all
received packages get processed. Previously packages from brokers would
not be processed while waiting for packages for other brokers, which
could run into the 1 sec timeout.
Sideeffect: some superfluous control packets will be sent in the
broker selection phase.

### CODE REVIEW

**Comment by kostko** [comment]
Tests seem to be failing due to a compile error in line 250, where you reference `ctx`, which is not defined:
```
FD_SET(ctx->fd, &rfds);
```

This line should just be removed as you configure fds later for all brokers.

File: client/l2tp_client.c [hunk 1]
```diff
@@ -205,6 +205,14 @@ typedef struct {
   time_t timer_resolving;
 } l2tp_context;
 
+// List of brokers
+typedef struct {
+    char *address;
```
Comment (kostko) [inline client/l2tp_client.c]: Fix indentation.

**Comment by kostko** [comment]
Something is strange with your last commit and Travis is getting confused for some reason and checking out a different commit, which is not actually there. The history shows "commited on GitHub", so this could be the reason.

Can you rebase your commits and push over here again?

**Comment by mitar** [comment]
This branch has conflicts anyway.

**Comment by Fungur69** [comment]
I merged the latest changes. I hope this builds now.

**Comment by mitar** [comment]
Tests are passing!

**Pull request merged**